### PR TITLE
Azure/acr content trust

### PR DESCRIPTION
**Pull request by alphadev4**

### CODE REVIEW

File: plugins/azure/containerregistry/acrContentTrustEnabled.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+            var found = false;
+            for (let registry of registries.data) {
+                if (!registry.id || (registry.sku && registry.sku.tier && registry.sku.tier!= 'Premium')) continue;
+
+                found = true;
+                var trustPolicy = registry.policies && registry.policies.trustPolicy? registry.policies.trustPolicy : null;
+
+                if (trustPolicy && trustPolicy.status && trustPolicy.status.toLowerCase() == 'enabled'){
+                    helpers.addResult(results, 0, 'Content trsut is enabled for container registry', location, registry.id);
```
Comment (alphadev4) [inline plugins/azure/containerregistry/acrContentTrustEnabled.js:45]:
```suggestion
                    helpers.addResult(results, 0, 'Content trust is enabled for container registry', location, registry.id);
```

File: plugins/azure/containerregistry/acrContentTrustEnabled.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                } else {
+                    helpers.addResult(results, 2, 'Content trsut is not enabled for container registry', location, registry.id);
```
Comment (alphadev4) [inline plugins/azure/containerregistry/acrContentTrustEnabled.js:47]:
```suggestion
                    helpers.addResult(results, 2, 'Content trust is not enabled for container registry', location, registry.id);
```

File: plugins/azure/containerregistry/acrContentTrustEnabled.spec.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                done()
+            });
+        });
+
+        it('should give failing result if content trsut is not enabled for container registry', function(done) {
+            const cache = createCache(null,[registries[0]]);
+            acrContentTrustEnabled.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).to.include('Content trsut is not enabled for container registry');
```
Comment (alphadev4) [inline plugins/azure/containerregistry/acrContentTrustEnabled.spec.js:80]:
```suggestion
                expect(results[0].message).to.include('Content trust is not enabled for container registry');
```

File: plugins/azure/containerregistry/acrContentTrustEnabled.spec.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            });
+
+        });
+
+        it('should give passing result if content trsut is enabled for container registry', function(done) {
+            const cache = createCache(null, [registries[1]]);
+            acrContentTrustEnabled.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('Content trsut is enabled for container registry');
```
Comment (alphadev4) [inline plugins/azure/containerregistry/acrContentTrustEnabled.spec.js:92]:
```suggestion
                expect(results[0].message).to.include('Content trust is enabled for container registry');
```

File: plugins/azure/containerregistry/acrContentTrustEnabled.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure/');
+
+module.exports = {
+    title: 'ACR Content Trust Enabled',
+    category: 'Container Registry',
+    domain: 'Containers',
+    description: 'Ensure that content trust is enabled for Azure premium container registries.',
+    more_info: 'Content trust allows you to sign the images you push to your registry. Consumers of your images (people or systems pulling images from your registry) can configure their clients to pull only signed images.',
```
Comment (mehakseedat63) [inline plugins/azure/containerregistry/acrContentTrustEnabled.js:9]:
```suggestion
    more_info: 'Content trust allows you to sign the images you push to your registry. Consumers of your images (people or systems pulling images from your registry) can configure their clients to pull only signed images which enhances container image security by ensuring the integrity and authenticity of images and safeguards against unauthorized or tampered content.',
```

File: plugins/azure/containerregistry/acrContentTrustEnabled.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            }
+
+            if (!registries.data.length) {
+                helpers.addResult(results, 0, 'No existing container registries found', location);
+                return rcb();
+            }
+
+            var found = false;
+            for (let registry of registries.data) {
+                if (!registry.id || (registry.sku && registry.sku.tier && registry.sku.tier!= 'Premium')) continue;
```
Comment (mehakseedat63) [inline plugins/azure/containerregistry/acrContentTrustEnabled.js:39]:
```suggestion
                if (!registry.id || (registry.sku && registry.sku.tier && registry.sku.tier!= 'Premium')) continue;
```
maybe add a pass result for non-premium registries that says 'is not a premium registry'

File: plugins/azure/containerregistry/acrContentTrustEnabled.spec.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+            acrContentTrustEnabled.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(3);
+                expect(results[0].message).to.include('Unable to query for container registries:');
+                expect(results[0].region).to.equal('eastus');
+                done()
+            });
+
+        });
+
```
Comment (mehakseedat63) [inline plugins/azure/containerregistry/acrContentTrustEnabled.spec.js:109]: add a test case for non-premium registries too

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**